src/App.tsx

```diff
@@ -1,26 +1,310 @@
+    setIsInputDisabled(true);
+
+    deleteTodo(todoID)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoID),
+        );
+        setIsInputDisabled(false);
+      })
+      .catch(() => {
+        setErrorDelete(true);
+
+        setTimeout(() => {
+          setErrorDelete(false);
+        }, 3000);
+      });
+  };
+
+  const handleDeleteCompletedTodos = () => {
+    setIsInputDisabled(true);
+
+    for (const todo of todos) {
+      if (todo.completed) {
+        deleteTodo(todo.id)
+          .then(() => {
+            setTodos(currentTodos =>
+              currentTodos.filter(currentTodo => currentTodo.id !== todo.id),
+            );
+          })
+          .catch(() => {
+            setErrorCompetedDelete(true);
+
+            setTimeout(() => {
+              setErrorCompetedDelete(false);
+            }, 3000);
+          })
+          .finally(() => {
+            setIsInputDisabled(false);
+          });
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (!isInputDisabled && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isInputDisabled]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this button should have `active` class only if all todos are completed */}
+          <button
+            type="button"
+            className={classNames('todoapp__toggle-all', {
+              active: todos.every(todo => todo.completed),
+              'is-hidden': !todos.length,
+            })}
+            data-cy="ToggleAllButton"
+          />
+
+          {/* Add a todo on form submit */}
+          <form onSubmit={handleSubmit}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              ref={inputRef}
+              value={title}
+              onChange={event => setTitle(event.target.value)}
+              disabled={isInputDisabled}
+            />
+          </form>
+        </header>
+
+        <TodoList
+          todos={filteredTodos}
+          onDeleteTodo={handleDeleteTodo}
+          tempTodo={tempTodo}
+          idDeletedTodo={idDeletedTodo}
+        />
+
+        {/* Hide the footer if there are no todos */}
+        {!todos.length || (
+          <footer className={classNames('todoapp__footer')} data-cy="Footer">
```

Comment:
also you should create component for Footer

Reply:
Good point